docs/typescript.md

```diff
@@ -0,0 +1,93 @@
+# Typescript
+
```

Comment:
Maybe worth adding a quick intro here, e.g. why we chose to migrate to TS & that we're actively in the process of migrating. e.g.

Our codebase is rapidly increasing in size and in contributors. This brings challenges to reliably keep the code organized, avoid code duplication, and review code. To help alleviate these challenges, we’re in the process of to TypeScript across our codebase. We believe having a strongly typed language will reduce bugs, improve code quality, increase productivity and allow us to scale (both our codebase and our developer community) better in the long term.